kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -96,6 +98,19 @@ func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispat
 	}
 	receiverFunc, err := eventingchannels.NewMessageReceiver(
 		func(ctx context.Context, channel eventingchannels.ChannelReference, message binding.Message, transformers []binding.Transformer, _ nethttp.Header) error {
+
+			// If we record the trace, we need to write the span info into the event
+			// using the distributed tracing extension.
+			// If the span is not recorded, we just don't send these info
+			span := trace.FromContext(ctx)
+			defer span.End()
```

Comment:
This should not attempt to end the existing span, it is the responsibility of the span creator to end the span.

Reply:
Since this span comes straight from the http middleware, if i remove it, I suppose it will close it